File: flowplayer-objc-demo/flowplayer-objc-demo/PlayerViewController.m
//
//  PlayerViewController.m
//  flowplayer-objc-demo
//
//  Created by Georgios Savvidis on 02.10.19.
//  Copyright © 2019 Flowplayer AB. All rights reserved.
//

#import "PlayerViewController.h"

@interface PlayerViewController () {}
@end

@implementation PlayerViewController

- (void)viewDidAppear:(BOOL)animated {
    _flowplayerViewController = [[FPFlowplayerViewController alloc] init];
    FPPlayerControlConfigBuilder * builder = [[FPPlayerControlConfig Builder] setMuteControl:true];
    builder = [builder enablePlugins:@[@"speed"]];
    builder = [builder setCustomWithKey:@"speed.options" value:@[@0.5, @1, @2, @5]];
    builder = [builder setCustomWithKey:@"speed.labels" value:@[@"Slow", @"Normal", @"Double", @"Fast"]];

    [_flowplayerViewController setControlConfig: [builder build]];

    [_containerView addSubview:_flowplayerViewController.view];

    // IMPORTANT: Add a delegate AFTER the player view has been added to your view.
    [_flowplayerViewController addPlayerDelegate:self];

    // IMPORTANT: Call prepare() AFTER the player view has been added to your view.
    if([self flowplayerMedia]) {
        [_flowplayerViewController prepareWithFlowplayerMedia:[self flowplayerMedia] autoStart:true];
    } else if ([self externalMedia]) {
        [_flowplayerViewController prepareWithExternalMedia:[self externalMedia] autoStart:true];
    }
}

- (void)onAdBreakCompleteWithEvent:(FPAdBreakCompleteEvent * _Nonnull)event {
    printf("OnAdBreakComplete");
}

- (void)onAdBreakStartWithEvent:(FPAdBreakStartEvent * _Nonnull)event {
    printf("OnAdBreakStart");
}

- (void)onAdClickWithEvent:(FPAdClickEvent * _Nonnull)event {
    printf("OnAdClick");
}

- (void)onAdCompleteWithEvent:(FPAdCompleteEvent * _Nonnull)event {
    printf("OnAdComplete");
}

- (void)onAdErrorWithEvent:(FPAdErrorEvent * _Nonnull)event {
    printf("OnAdError");
}

- (void)onAdPauseWithEvent:(FPAdPauseEvent * _Nonnull)event {
    printf("OnAdPause");
}

- (void)onAdResumeWithEvent:(FPAdResumeEvent * _Nonnull)event {
    printf("OnAdResume");
}

- (void)onAdSkipWithEvent:(FPAdSkipEvent * _Nonnull)event {
    printf("OnAdSkip");
}

- (void)onAdStartWithEvent:(FPAdStartEvent * _Nonnull)event {
    printf("OnAdStart");
}

- (void)onBufferWithEvent:(FPBufferEvent * _Nonnull)event {
    printf("OnBuffer");
}

- (void)onCompleteWithEvent:(FPCompleteEvent * _Nonnull)event {
    printf("OnComplete");
}

- (void)onErrorWithEvent:(FPErrorEvent * _Nonnull)event {
    printf("OnError");
}

- (void)onFullscreenWithEvent:(FPFullscreenEvent * _Nonnull)event {
    printf("OnFullscreen");
}

- (void)onMuteWithEvent:(FPMuteEvent * _Nonnull)event {
    printf("OnMuteChanged");
}

- (void)onVolumeWithEvent:(FPVolumeEvent * _Nonnull)event {
    printf("OnVolumeChanged");
}

- (void)onIdleWithEvent:(FPIdleEvent * _Nonnull)event {
    printf("OnIdle");
}

- (void)onPauseWithEvent:(FPPauseEvent * _Nonnull)event {
    printf("OnPause");
}

- (void)onPlayWithEvent:(FPPlayEvent * _Nonnull)event {
    printf("OnPlay");
}

- (void)onSpeedWithEvent:(FPSpeedEvent * _Nonnull)event {
    printf("OnSpeed");
}

- (void)onAudioTracksWithEvent:(FPAudioTracksEvent * _Nonnull)event {
    printf("OnAudioTracks");
}

- (void)onAudioTrackSelectWithEvent:(FPAudioTrackSelectEvent * _Nonnull)event {
    printf("OnAudioTrackSelect");
}

- (void)onOvpMetadataWithEvent:(FPOvpMetadataEvent * _Nonnull)event {
    printf("OnOvpMetadata");
}

- (void)onCastingWithEvent:(FPCastingEvent * _Nonnull)event {
	printf("onCastingWithEvent");
}


- (void)onSubtitleTrackSelectWithEvent:(FPSubtitleTrackSelectEvent * _Nonnull)event {
	printf("onSubtitleTrackSelectWithEvent");
}


- (void)onSubtitleTracksWithEvent:(FPSubtitleTracksEvent * _Nonnull)event {
	printf("onSubtitleTracksWithEvent");
}


- (void)encodeWithCoder:(nonnull NSCoder *)coder {
	printf("encodeWithCoder");
}

- (void)traitCollectionDidChange:(nullable UITraitCollection *)previousTraitCollection {
	printf("traitCollectionDidChange");
}

- (void)preferredContentSizeDidChangeForChildContentContainer:(nonnull id<UIContentContainer>)container {
	printf("preferredContentSizeDidChangeForChildContentContainer");
}

- (CGSize)sizeForChildContentContainer:(nonnull id<UIContentContainer>)container withParentContainerSize:(CGSize)parentSize {
	printf("sizeForChildContentContainer");
	return CGSizeZero;
}

- (void)systemLayoutFittingSizeDidChangeForChildContentContainer:(nonnull id<UIContentContainer>)container {
	printf("systemLayoutFittingSizeDidChangeForChildContentContainer");
}

- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(nonnull id<UIViewControllerTransitionCoordinator>)coordinator {
	printf("viewWillTransitionToSize");
}

- (void)willTransitionToTraitCollection:(nonnull UITraitCollection *)newCollection withTransitionCoordinator:(nonnull id<UIViewControllerTransitionCoordinator>)coordinator {
	printf("viewWillTransitionToSize");
}

- (void)didUpdateFocusInContext:(nonnull UIFocusUpdateContext *)context withAnimationCoordinator:(nonnull UIFocusAnimationCoordinator *)coordinator {
	printf("didUpdateFocusInContext");
}

- (void)setNeedsFocusUpdate {
	printf("setNeedsFocusUpdate");
}

- (BOOL)shouldUpdateFocusInContext:(nonnull UIFocusUpdateContext *)context {
	printf("shouldUpdateFocusInContext");
	return false;
}

- (void)updateFocusIfNeeded {
	printf("updateFocusIfNeeded");
}

@end

File: flowplayer-demo/PlayerViewController.swift
//
//  ViewController.swift
//  flowplayer-demo
//
//  Created by Georgios Savvidis on 01.09.19.
//  Copyright © 2019 Flowplayer AB. All rights reserved.
//

import UIKit
import FlowplayerCore

class PlayerViewController: UIViewController, FPFlowplayerDelegate {
    @IBOutlet weak var containerView: UIView!
    
    var flowplayerMedia: FPFlowplayerMedia?
    var externalMedia: FPExternalMedia?

    private let flowplayerViewController = FPFlowplayerViewController()

    override func viewDidAppear(_ animated: Bool) {
        
        let config = FPPlayerControlConfigBuilder()
            .setMuteControl(true)
            .enablePlugins(["speed"])
            .setCustom(key: "speed.options", value: [0.5, 1, 2, 5])
            .setCustom(key: "speed.labels", value: ["Slow", "Normal", "Double", "Fast"])
            .build()
        
        flowplayerViewController.setControlConfig(config)
        
        containerView.addSubview(flowplayerViewController.view)

        // Add the delegate after the flowplayerViewController has been added to the container.
        flowplayerViewController.addPlayerDelegate(self)

        if let flowplayerMedia = flowplayerMedia {
            flowplayerViewController.prepare(flowplayerMedia: flowplayerMedia, autoStart: true)
        } else if let externalMedia = externalMedia {
            flowplayerViewController.prepare(externalMedia: externalMedia, autoStart: true)
        }
    }
    
    func onIdle(event: FPIdleEvent) {
        print("OnIdle")
    }
    
    func onPlay(event: FPPlayEvent) {
        print("OnPlay")
    }
    
    func onPause(event: FPPauseEvent) {
        print("OnPause")
    }
    
    func onBuffer(event: FPBufferEvent) {
        print("OnBuffer")
    }
    
    func onComplete(event: FPCompleteEvent) {
        print("OnComplete")
    }
    
    func onFullscreen(event: FPFullscreenEvent) {
        print("OnFullscreen")
    }
  
    func onMute(event: FPMuteEvent) {
        print("OnMute")
    }
    
    func onVolume(event: FPVolumeEvent) {
        print("OnVolume")
    }
    
    func onSpeed(event: FPSpeedEvent) {
        print("OnSpeed: \(event.speed)")
    }
    
    func onError(event: FPErrorEvent) {
        print("OnError")
    }
    
    func onAdBreakStart(event: FPAdBreakStartEvent) {
        print("OnAdBreakStart")
    }
    
    func onAdBreakComplete(event: FPAdBreakCompleteEvent) {
        print("OnAdBreakComplete")
    }
    
    func onAdStart(event: FPAdStartEvent) {
        print("OnAdStart")
    }
    
    func onAdPause(event: FPAdPauseEvent) {
        print("OnAdPause")
    }
    
    func onAdResume(event: FPAdResumeEvent) {
        print("OnAdResume")
    }
    
    func onAdClick(event: FPAdClickEvent) {
        print("OnAdClick")
    }
    
    func onAdSkip(event: FPAdSkipEvent) {
        print("OnAdSkip")
    }
    
    func onAdComplete(event: FPAdCompleteEvent) {
        print("OnAdComplete")
    }
    
    func onAdError(event: FPAdErrorEvent) {
        print("OnAdError")
    }
}

